torch/linalg/__init__.py

```diff
@@ -344,49 +381,78 @@
+The eigenvectors will have the same dtype as :attr:`input`.
 
-.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+.. note:: When :attr:`input` is on a CUDA device, this function synchronizes that device with the CPU.
 
-.. note:: The eigenvalues/eigenvectors are computed using LAPACK's `syevd` and `heevd` routines for CPU inputs,
-          and MAGMA's `syevd` and `heevd` routines for CUDA inputs.
+.. note:: This function is computed using LAPACK's and MAGMA's `syevd` and `heevd` routines for
+          CPU and CUDA symmetric and Hermitian inputs respectively.
 
 .. note:: The eigenvalues of real symmetric or complex Hermitian matrices are always real.
 
-.. note:: The eigenvectors of matrices are not unique, so any eigenvector multiplied by a constant remains
-          a valid eigenvector. This function may compute different eigenvector representations on
-          different device types. Usually the difference is only in the sign of the eigenvector.
+.. warning:: The eigenvectors of a matrix are not unique. Any eigenvector may be multiplied by `-1`
+             in the real case or by :math:`e^{i \phi}` for any :math:`\phi \in \mathbb{R}` in the
+             complex case.
+             As such, the eigenvectors are not continuous functions of the inputs, and different
+             platforms, like NumPy, or inputs on different devices, may produce different
+             eigenvectors.
 
-.. note:: See :func:`torch.linalg.eigvalsh` for a related function that computes only eigenvalues.
-          However, that function is not differentiable.
+.. warning:: If `V` is used in the loss function, the gradient will only be finite when
+             :attr:`input` does not have repeated eigenvalues.
+             If the distance between any two eigenvalues is close to zero, the gradient with
+             respect to `V` will be numerically unstable, as it depends on
+             :math:`\frac{1}{\min_{i \neq j} \lambda_i - \lambda_j}`.
 
-Args:
-    input (Tensor): the Hermitian `n \times n` matrix or the batch of such matrices of size
-                    `(*, n, n)` where `*` is one or more batch dimensions.
-    UPLO ('L', 'U', optional): controls whether to use the upper-triangular or the lower-triangular part
-                               of :attr:`input` in the computations. Default is ``'L'``.
+.. seealso::
 
-Keyword args:
-    out (tuple, optional): tuple of two tensors to write the output to. Default is ``None``.
+        :func:`torch.linalg.eigvalsh` for a related function that computes only eigenvalues.
+        However, that function is not differentiable.
 
-Returns:
-    (Tensor, Tensor): A namedtuple (eigenvalues, eigenvectors) containing
+        :func:`torch.linalg.cholesky` for a different decomposition of a Hermitian matrix.
+        The Cholesky decomposition gives less information about the matrix but it is much faster
```

Comment:
How comfortable are we making definitive statements about speed? Will a user complain if our cholesky decomposition is ever slower because we're using some goofball math library to compute it?

Reply:
I think it's great to put these, and if anything, we should try to make them true.

This is what the theory says, and PyTorch should offer similar guarantees. I also just put the very obvious partial order:
cholesky ~ qr < eigh < eig ~ svd.

If any of those strict comparisons do not hold, we should feel very bad and really try to fix it in my opinion.

This comment also goes in the line of my secret plan of teaching the user linear algebra while reading the docs, along the lines to the comments linking `inv` with `solve` and `pinv` with `lstsq`.